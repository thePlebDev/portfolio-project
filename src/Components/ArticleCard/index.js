import React from 'react';
import styled from 'styled-components';

import Filter from '../Filters'

const Container = styled.div`
text-align:center;
margin-bottom:20px;
`

const Description = styled.div`
  width:350px;
  margin:1rem auto;
  padding:10px;
  font-size: 1rem;
  line-height: 1.75rem;
`
const Title = styled.div`
  font-size:1.2em;
  margin-top:1em;
  position:relative;
  ::after{
    content:'';

    width:80px;
    height:2px;
    background-color:red;
    position:absolute;
    left:0;
    bottom:0;
    right:0;
    margin:-5px auto;
  }
`

const ReadMore = styled.div`
  font-size: 0.75rem;
`

const FilterContainer =styled.div`
  display:flex;
  align-items:center;
  justify-content:center;
`

const ArticleCard =({image,title,filters,description,setBlogState,blogState})=>{

  return(
    <Container>
      <img  style={{height:'200px',width:'350px'}} src={image} alt='city' />
      <Filter filters={filters} setBlogState={setBlogState} blogState={blogState}/>
      <Title>{title}</Title>
      <Description>{description}</Description>
      <ReadMore> Read more</ReadMore>
    </Container>
  )

}

export default ArticleCard
