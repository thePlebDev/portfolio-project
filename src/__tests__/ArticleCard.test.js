import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';
import ArticleCard from '../Components/ArticleCard'
import {BrowserRouter as Router} from 'react-router-dom'

//1) check the snapshot
//2) check that it renders properly
//3) check the default props
//4) check the custom props
//5) check the events


describe('Testing the ArticleCard component',()=>{
  let wrapper;
  beforeEach(()=>{
    wrapper = shallow(<ArticleCard />)
  })
  it('should create the snapshot',()=>{
    const tree = renderer.create(<Router><ArticleCard/></Router>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should render all the important stuff',()=>{

    expect(wrapper.find('[data-testid="image"]').length).toBe(1)
    expect(wrapper.find('[data-testid="title"]').length).toBe(1)
    expect(wrapper.find('[data-testid="description"]').length).toBe(1)
    expect(wrapper.find('[data-testid="read-more"]').length).toBe(1)
  })
  it('should render the default prop values',()=>{
    const defaultImage="https://images.unsplash.com/photo-1477959858617-67f85cf4f1df?ixlib=rb-1.2.1&auto=format&fit=crop&w=1713&q=80"
    const defaultDescription="Click read more to view the whole article"
    const defaultTitle="This post deals with JavaScript"
    
    expect(wrapper.find('[data-testid="image"]').props().src).toBe(defaultImage)
    expect(wrapper.find('[data-testid="title"]').props().children).toBe(defaultTitle)
    expect(wrapper.find('[data-testid="description"]').props().children).toBe(defaultDescription)

  })
})
