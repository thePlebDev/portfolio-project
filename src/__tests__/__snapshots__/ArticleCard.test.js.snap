// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the ArticleCard component should create the snapshot 1`] = `
<div
  className="sc-dlnjPT kYIEqD"
>
  <img
    alt="city"
    data-testid="image"
    src="https://images.unsplash.com/photo-1477959858617-67f85cf4f1df?ixlib=rb-1.2.1&auto=format&fit=crop&w=1713&q=80"
    style={
      Object {
        "height": "200px",
        "width": "350px",
      }
    }
  />
  <div
    className="sc-bdnylx eMLUIs"
  >
    <div
      className="sc-gtssRu bENsCd"
      onClick={[Function]}
    >
      #React
    </div>
  </div>
  <div
    className="sc-eCApGN bKfbFH"
    data-testid="title"
  >
    This post deals with JavaScript
  </div>
  <div
    className="sc-hKFyIo bOhjDH"
    data-testid="description"
  >
    Click read more to view the whole article
  </div>
  <a
    href="/blog/undefined"
    onClick={[Function]}
  >
    <div
      className="sc-jSFkmK hJsiHT"
      data-testid="read-more"
    >
       Read more
    </div>
  </a>
</div>
`;
